actions/trigger-argo-workflow/cmd/trigger-argo-workflow/main.go

```diff
@@ -69,35 +76,64 @@ func main() {
 		)
 	}
 
-	app := cli.NewApp()
+	app := cli.Command{}
+	app.Writer = writer
+	app.ErrWriter = errWriter
 	app.Name = "Runs the Argo CLI"
 
-	app.Action = func(c *cli.Context) error { return run(c, &lv, logger) }
+	app.Action = func(ctx context.Context, c *cli.Command) error {
+		return fmt.Errorf("please specify a command")
+	}
+
+	app.Commands = []*cli.Command{
+		{
+			Name:            "submit",
+			SkipFlagParsing: true,
+			Writer:          writer,
+			ErrWriter:       errWriter,
+			Action: func(ctx context.Context, c *cli.Command) error {
+				return run(ctx, c, &lv, logger, "submit")
+			},
+		},
+	}
 
 	app.Flags = []cli.Flag{
 		&cli.BoolFlag{
-			Name:    flagAddCILabels,
-			EnvVars: []string{"ADD_CI_LABELS"},
-			Value:   false,
-			Usage:   "If true, the `--labels` argument will be added with values from the environment. This is forced for the `submit` command",
+			Name:     flagPrintConfig,
+			Required: false,
+			Usage:    "If set thie command will only print the gathered configuration and exist",
+		},
+		&cli.BoolFlag{
+			Name: flagAddCILabels,
+			Sources: cli.NewValueSourceChain(
+				cli.EnvVar("ADD_CI_LABELS"),
+			),
```

Comment:
q: Over on [the migration guide](https://cli.urfave.org/migrate-v2-to-v3/#__tabbed_9_2), they suggest a shorter form. Would that work?

Reply:
That would also work. Personally, I think the longer form is just more readable as it sets `Sources` to something that hints at a list of sources 🙂